tests/testthat/test-pivot_wider_pknca_results.R

```diff
@@ -0,0 +1,165 @@
+
+  # Extract unique parameter names with units
+  names_params <- TEST_PKNCA_RES$result %>%
+    mutate(
+      PPSTRESU2 = ifelse(
+        PPSTRESU %in% c("", "unitless", "fraction"),
+        "",
+        paste0("[", PPSTRESU, "]")
+      ),
+      PPTESTCD2 = ifelse(
+        type_interval == "manual",
+        paste0(PPTESTCD, "_", start, "-", end, PPSTRESU2),
+        paste0(PPTESTCD, PPSTRESU2)
+      ),
+    ) %>%
+    pull(PPTESTCD2) %>%
+    unique()
+
+  # Iterate over each parameter and validate values
+  for (param_col in names_params) {
+    pptestcd <- gsub("\\[.*\\]", "", param_col)  # Remove units from parameter name
+
+    if (grepl("AUCINT_[0-9]+\\-[0-9]", param_col)) {
+      auc_start <- gsub("AUCINT_([0-9]+)\\-[0-9]", "\\1", pptestcd)
+      auc_end <- gsub("AUCINT_[0-9]+\\-([0-9]+)", "\\1", pptestcd)
+      pptestcd <- gsub("_[0-9]+\\-[0-9]", "", pptestcd)
+
+      original_data <- TEST_PKNCA_RES$result %>%
+        filter(start == auc_start,
+               end == auc_end)
+      reshaped_data <- pwres %>%
+        filter(!is.na(!!sym(param_col)))
+    } else {
+      original_data <- TEST_PKNCA_RES$result
+      reshaped_data <- pwres
+    }
+
+    original_vals <- original_data %>%
+      formatters::var_labels_remove() %>%
+      filter(PPTESTCD == pptestcd) %>%
+      pull(PPSTRES)
+    result_vals <- reshaped_data %>%
+      formatters::var_labels_remove() %>%
+      pull(param_col)
+
+    have_similar_vals <- all(round(result_vals, 3) %in% round(original_vals, 3))
+    have_same_length <- length(result_vals) == length(original_vals)
+    if (!(have_similar_vals && have_same_length)) {
+      stop(paste("Mismatch in values for parameter:", param_col))
+    }
+  }
+
+  # If all checks pass, return TRUE
+  TRUE
+}
+
+
+describe("pivot_wider_pknca_results", {
+
+  it("produces a data.frame with expected format when only reshaping main intervals", {
+    res_only_main <- TEST_PKNCA_RES
+    res_only_main$result <- TEST_PKNCA_RES$result  %>%
+      filter(type_interval == "main")
+    pwres_only_main <- pivot_wider_pknca_results(res_only_main)
+    expect_s3_class(pwres_only_main, "data.frame")
+
+    # Check that the result contains expected columns
+    expected_columns <- c(
+      "USUBJID", "start", "end", "DOSNO", "AFRLT", "ARRLT",
+      "NFRLT", "NRRLT", "ADOSE", "ROUTE", "ADOSEDUR",
+      "CMAX[ng/mL]", "TMAX[hr]", "TLST[hr]", "LAMZ[1/hr]",
+      "R2", "R2ADJ", "LAMZLL[hr]",
+      "LAMZNPT[count]", "CLSTP[ng/mL]", "LAMZHL[hr]",
+      "LAMZSPN", "LAMZIX",
+      "LAMZMTD", "Exclude"
+    )
+    expect_true(all(expected_columns %in% colnames(pwres_only_main)))
+
+    # Check number of rows is the expected (each dose has 1 row in the pivoted data)
+    expect_true(nrow(pwres_only_main) == nrow(res_only_main$data$dose$data))
+
+    # Check parameter values match the ones in the PKNCA results object
+    expect_true(.validate_pknca_params(res_only_main, pwres_only_main))
```

Comment:
issue: Should be more like
```suggestion
    expect_no_error(.validate_pknca_params(res_only_main, pwres_only_main))
```
since if there is an issue, the validate function will throw an error - and this is what we want to have reported. Then the validate function can return whatever (or nothing) and the test will still be as informative. The returned value is not relevant in the context of testing - the errors are.